test/new-e2e/pkg/environments/aws/eks/eks.go

```diff
@@ -0,0 +1,414 @@
+func WithExtraConfigParams(configMap runner.ConfigMap) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.extraConfigParams = configMap
+		return nil
+	}
+}
+
+// WithFakeIntakeOptions sets the options for the FakeIntake
+func WithFakeIntakeOptions(opts ...fakeintake.Option) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.fakeintakeOptions = append(params.fakeintakeOptions, opts...)
+		return nil
+	}
+}
+
+// WithEKSLinuxNodeGroup enable aws/ecs/linuxECSOptimizedNodeGroup
+func WithEKSLinuxNodeGroup() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.eksLinuxNodeGroup = true
+		return nil
+	}
+}
+
+// WithEKSLinuxARMNodeGroup enable aws/ecs/linuxECSOptimizedNodeGroup
+func WithEKSLinuxARMNodeGroup() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.eksLinuxARMNodeGroup = true
+		return nil
+	}
+}
+
+// WithEKSBottlerocketNodeGroup enable aws/ecs/linuxECSOptimizedNodeGroup
+func WithEKSBottlerocketNodeGroup() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.eksBottlerocketNodeGroup = true
+		return nil
+	}
+}
+
+// WithECSWindowsNodeGroup enable aws/ecs/windowsLTSCNodeGroup
+func WithEKSWindowsNodeGroup() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.eksWindowsNodeGroup = true
+		return nil
+	}
+}
+
+// WithDeployDogstatsd deploy standalone dogstatd
+func WithDeployDogstatsd() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.deployDogstatsd = true
+		return nil
+	}
+}
+
+// WithoutFakeIntake deactivates the creation of the FakeIntake
+func WithoutFakeIntake() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.fakeintakeOptions = nil
+		return nil
+	}
+}
+
+// WithoutAgent deactivates the creation of the Docker Agent
+func WithoutAgent() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.agentOptions = nil
+		return nil
+	}
+}
+
+// WorkloadAppFunc is a function that deploys a workload app to a kube provider
+type WorkloadAppFunc func(e config.CommonEnvironment, kubeProvider *kubernetes.Provider) (*kubeComp.Workload, error)
+
+// WithWorkloadApp adds a workload app to the environment
+func WithWorkloadApp(appFunc WorkloadAppFunc) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.workloadAppFuncs = append(params.workloadAppFuncs, appFunc)
+		return nil
+	}
+}
+
+// Run deploys a EKS environment given a pulumi.Context
+func Run(ctx *pulumi.Context, env *environments.EKS, params *ProvisionerParams) error {
+	var awsEnv aws.Environment
+	var err error
+	if env.AwsEnvironment != nil {
+		awsEnv = *env.AwsEnvironment
+	} else {
+		awsEnv, err = aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	clusterComp, err := components.NewComponent(*awsEnv.CommonEnvironment, awsEnv.Namer.ResourceName("eks"), func(comp *kubeComp.Cluster) error {
```

Comment:
I think this component should live in `test-infra-definitions`. Do you think you could create it on `test-infra-definitions` side. We have a similar component for `Kind` cluster here: https://github.com/DataDog/test-infra-definitions/blob/c5f8da77ddbc87908ce7cc397b7b8ccd8c62ddd6/components/kubernetes/kind.go#L29

Reply:
Indeed, we should have components and use them in custom provisioner rather than forcing the standard provisioners.

Reply:
As discussed offline with @pducolin I moved eks.go in aws/kubernetes and this will create a component